### PR TITLE
[Arm64] Fix GC Holes in emitInsLoadStoreOp

### PR DESCRIPTION
**Pull request by sdmaclea**

### CODE REVIEW

**Comment by sdmaclea** [comment]
@dotnet/arm64-contrib

File: src/jit/emitarm64.cpp [hunk 1]
```diff
@@ -10820,13 +10820,13 @@ void emitter::emitInsLoadStoreOp(instruction ins, emitAttr attr, regNumber dataR
                     if (lsl > 0)
                     {
                         // Generate code to set tmpReg = base + index*scale
-                        emitIns_R_R_R_I(INS_add, EA_PTRSIZE, tmpReg, memBase->gtRegNum, index->gtRegNum, lsl,
+                        emitIns_R_R_R_I(INS_add, EA_BYREF, tmpReg, memBase->gtRegNum, index->gtRegNum, lsl,
```
Comment (sdmaclea) [inline src/jit/emitarm64.cpp]: This is making the assumption that the base register is referring to a GC object.  I probably need to check a flag on memBase...

Same below.

Comment (briansull) [inline src/jit/emitarm64.cpp]: `emitAttr addType      =  varTypeIsGC(memBase) ? EA_BYREF : EA_PTRSIZE;`
...
`emitIns_R_R_R_I(INS_add, addType, tmpReg, memBase->gtRegNum, index->gtRegNum, lsl,`

File: src/jit/emitarm64.cpp [hunk 2]
```diff
@@ -10820,13 +10820,13 @@ void emitter::emitInsLoadStoreOp(instruction ins, emitAttr attr, regNumber dataR
...
                                         INS_OPTS_LSL);
                     }
                     else // no scale
                     {
                         // Generate code to set tmpReg = base + index
-                        emitIns_R_R_R(INS_add, EA_PTRSIZE, tmpReg, memBase->gtRegNum, index->gtRegNum);
+                        emitIns_R_R_R(INS_add, EA_BYREF, tmpReg, memBase->gtRegNum, index->gtRegNum);
```
Comment (briansull) [inline src/jit/emitarm64.cpp]: emitIns_R_R_R(INS_add, addType, tmpReg, memBase->gtRegNum, index->gtRegNum);

File: src/jit/emitarm64.cpp [hunk 3]
```diff
@@ -10841,7 +10841,7 @@ void emitter::emitInsLoadStoreOp(instruction ins, emitAttr attr, regNumber dataR
                     codeGen->instGen_Set_Reg_To_Imm(EA_PTRSIZE, tmpReg, offset);
                     // Then add the base register
                     //      rd = rd + base
-                    emitIns_R_R_R(INS_add, EA_PTRSIZE, tmpReg, tmpReg, memBase->gtRegNum);
+                    emitIns_R_R_R(INS_add, EA_BYREF, tmpReg, tmpReg, memBase->gtRegNum);
```
Comment (briansull) [inline src/jit/emitarm64.cpp]: `emitIns_R_R_R(INS_add, addType, tmpReg, tmpReg, memBase->gtRegNum);`

**Comment by sdmaclea** [comment]
@briansull Thanks.  Pushed a patch to fix.

**Comment by jashook** [comment]
@dotnet-bot test Windows_NT Arm64 Checked

**Comment by sdmaclea** [comment]
@jashook Looks like the arm64 build timed out (possibly waiting for tests to build?)

**Comment by jashook** [comment]
@dotnet-bot test Windows_NT Arm64 Checked

**Comment by sdmaclea** [comment]
@dotnet-bot test Tizen armel Cross Release Build please

**Comment by briansull** [comment]
LGTM

**Comment by jashook** [comment]
@dotnet-bot test Windows_NT Arm64 Checked

**Comment by BruceForstall** [comment]
@dotnet-bot test Windows_NT Arm64 Checked
@dotnet-bot test Tizen armel Cross Release Build

**Comment by sdmaclea** [comment]
@dotnet/arm64-contrib Looks like the last arm64 run died due to network timeout.  Can someone retrigger it if/when arm64 CI is stable

**Comment by sdmaclea** [comment]
@dotnet/arm64-contrib ping ---> test Windows_NT Arm64 Checked please

**Comment by jashook** [comment]
@dotnet-bot test Windows_NT Arm64 Checked

**Pull request merged**